builders/ethereum/precompiles/utility/non-specific.md

```diff
@@ -0,0 +1,91 @@
+
+In the following sections, you will learn how to use the `ECRecoverPublicKey` precompile.
+
+### Checking Prerequisites {: checking-prerequisites }
+
+--8<-- 'text/_common/install-nodejs.md'
+
+The versions used in this example are v20.15.0 (Node.js) and 10.7.0 (npm). You will also need to install the [Web3](https://web3js.readthedocs.io/en/latest){target=\_blank} package by executing:
+
+```bash
+npm install --save web3
+```
+
+To verify the installed version of Web3, you can use the `ls` command:
+
+```bash
+npm ls web3
+```
+
+This example uses version 4.11.1. You will also use [Remix](/builders/ethereum/dev-env/remix/){target=\_blank}, connecting it to the Moonbase Alpha TestNet via [MetaMask](/tokens/connect/metamask/){target=\_blank}.
+
+--8<-- 'text/_common/endpoint-examples.md'
+
+### Retrieve Transaction Signature Values
+
+To use the `ECRecoverPublicKey` precompile, you must first sign a message to create and retrieve the message hash and transaction signature values (`v`, `r`, `s`) to pass as arguments in the contract call. Please note that the address and private key values used here are hard-coded for simplicity. Always use security best practices when handling private keys. 
+
+Create a new file called `signMessage.js` in your project directory:
+
+```bash
+touch signMessage.js
+```
+
+Open `signMessage.js` in your code editor and add the following script to initialize Web3 with Moonbase Alpha TestNet, sign and hash the message, and return the signature values:
+
+```js title="signMessage.js"
+--8<-- 'code/builders/ethereum/precompiles/utility/nonspecific/signMessage.js'
+```
+
+Return to your terminal command line to run the script with this command:
+
+```bash
+node signMessage.js
+```
+
+This code will return the following object in the terminal:
+
+--8<-- 'code/builders/ethereum/precompiles/utility/nonspecific/terminal/signature.md'
+
+Save these values as you will need them in the next section.
+
+### Test ECRecoverPublicKey Contract
+
+You can now visit [Remix](/builders/ethereum/dev-env/remix/){target=\_blank} to test the precompiled contract. Note that you could also use the Web3.js library, but in this case, you can go to Remix to ensure it is using the precompiled contract on the blockchain. The Solidity code you can use to retrieve the public key is the following:
+
+```solidity title="RecoverPublicKey.sol"
+--8<-- 'code/builders/ethereum/precompiles/utility/nonspecific/RecoverPublicKey.sol'
+```
+
+Using the [Remix compiler and deployment](/builders/ethereum/dev-env/remix/){target=\_blank} and with [MetaMask pointing to Moonbase Alpha](/tokens/connect/metamask/){target=\_blank}, you can deploy the contract and call the `recoverPublicKey()` method which returns the public key for the account that signed the message. You can now use this public key value for other cryptographic functions and verifications.
+
+![Returned Public Key on Remix](/images/builders/ethereum/precompiles/utility/nonspecific/nonspecific-1.webp)
```

Comment:
Unless our image standards have changed to have small image dimensions, I think this one is a tad small. I have a large monitor, but this is how it looks on my screen:
<img width="2010" alt="Screenshot 2024-08-15 at 4 15 13 PM" src="https://github.com/user-attachments/assets/e7f9d652-fbc1-4f5c-a2c9-5d5248400527">

Reply:
If it's possible to take this screenshot on a larger window I think that would be helpful

Reply:
Wait but where is your actual browser window? @themacexpert

Reply:
I did say to resize images to 760px so they aren't unnecessarily too big because I didn't think the width of the image would get much wider on large screens. I thought just the left and right margins of the page would be wider. Can you send you're entire screen with left nav and toc in screenshot too please?

Reply:
Yeah, this is 756px wide per the Docs Standards page in Confluence. Let me know if I need to do something else.

Reply:
Spun this up locally and viewed in Chrome Dev Tools with ~1400 px wide screen
![fullscreen](https://github.com/user-attachments/assets/00627680-9168-42dc-8895-6fab22783d1c)

Reply:
Here's a view with the whole window after clicking on the image to enlarge. 

<img width="2015" alt="Screenshot 2024-08-16 at 8 37 47 AM" src="https://github.com/user-attachments/assets/39d45360-98d0-435f-961a-c312a20eb68f">

Reply:
I personally don't think it matters if it fills up the whole screen when opening the image in a new tab, it's really about how it's viewed on the actual content page. We should be using [properly sized images](https://developer.chrome.com/docs/lighthouse/performance/uses-responsive-images). 

But with this all being said, the screenshot is a bit blurry. Luca played around with doubling the width to 1512px for some Polkadot-stuff and said that turned out better for him and didn't increase the kb of the image too much. So just try that